miio/command_sender.py

```diff
@@ -0,0 +1,235 @@
+import binascii
+import codecs
+import datetime
+import logging
+import socket
+from typing import Any, List  # noqa: F401
+
+import construct
+
+from .exceptions import DeviceError, DeviceException, RecoverableError
+from .protocol import Message
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class CommandSender:
```

Comment:
Rename this to `Protocol`.

Reply:
I started to wonder if this should be under `miioprotocol.py` (maybe current `protocol.py` is misnamed...) considering you are going to add `miotprotocol.py` in your another PR, no?

This way the pieces shared by both protocols would live separately from the specific implementation. What do you think?

Reply:
yep, we'll probably end up with these 3 classes. maybe we can do the split in the next PR?

Reply:
Ok, fair enough :+1: